homeassistant/components/websocket_api/connection.py

```diff
@@ -72,12 +73,12 @@ def send_error(self, msg_id: int, code: str, message: str) -> None:
     @callback
     def async_handle(self, msg: dict[str, Any]) -> None:
         """Handle a single incoming message."""
-        handlers = self.hass.data[const.DOMAIN]
-
-        try:
-            msg = messages.MINIMAL_MESSAGE_SCHEMA(msg)
-            cur_id = msg["id"]
-        except vol.Invalid:
+        if (
+            not (cur_id := msg.get("id"))
```

Comment:
we should also validate that it's a dict.
```suggestion
            # Not using isinstance as we don't care about children
            type(msg) is dict and
            not (cur_id := msg.get("id"))
```